packages/types/src/options.ts

```diff
@@ -106,6 +107,11 @@ export interface Options {
    */
   shutdownTimeout?: number;
 
+  /**
+   * Set of metadata about the SDK that can be used to enhance envelopes/events and provide additional data about every request.
+   * */
+  metadata?: Partial<SdkInfo>;
```

Comment:
```suggestion
  metadata?: SdkInfo;
```

Why make this a partial? Everything in `SdkInfo` besides `name` and `version` is already optional. And if we were to include package info in the various `init` methods, it would make it easier to get rid of all of those boilerplate event processors. (We could add the integrations metadata to the `Api` object as part of the browser and node `init` methods upon which the other `init`s depend, which would fulfill your TODO of getting rid of `applyIntegrationsMetadata` as well.)

Reply:
Because without `Partial`, I'd need to construct `{ version: 'to_be_overwritten_by_core', name: 'sentry.javascript.browser' }` in all `init` calls.

Reply:
> And if we were to include package info in the various init methods, it would make it easier to get rid of all of those boilerplate event processors. (We could add the integrations metadata to the Api object as part of core.sdk.initAndBind, which would fulfill your TODO of getting rid of applyIntegrationsMetadata as well.)

Let me tinker with it.